vars/buildDockerAndPublishImage.groovy

```diff
@@ -138,7 +138,7 @@ def call(String imageName, Map userConfig=[:]) {
             usernamePassword(credentialsId: "${finalConfig.gitCredentials}", passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')
           ]) {
             echo "Tagging New Version: $nextVersion"
-            sh "git tag $nextVersion"
+            sh "git tag -a $nextVersion -m $IMAGE_NAME"
```

Comment:
Good idea to add a commit message with the image name!

But you have to also update the tests (check for `-a` flag + you have to mock the $IMAGE_NAME` variable in the sh mock)